theano/compile/tests/test_function_module.py

```diff
@@ -239,6 +239,41 @@ def test_copy(self):
         f(1, 2)  # put them out of sync
         self.assertFalse(f(1, 2) == g(1, 2))  # they should not be equal anymore.
 
+    def test_copy_share_memory(self):
```

Comment:
You had a tests with shared variable. Add it in another test. We need 2 tests, one where we don't update the shared variable and one where we do.